lib/HttpClient.ts

```diff
@@ -105,6 +106,14 @@ export class HttpClient implements ifm.IHttpClient {
     constructor(userAgent: string, handlers?: ifm.IRequestHandler[], requestOptions?: ifm.IRequestOptions) {
         this.userAgent = userAgent;
         this.handlers = handlers || [];
+        let no_proxy: string = process.env[EnvironmentVariables.NO_PROXY];
+        if (no_proxy) {
+            this._httpProxyBypassHosts = [];
+            no_proxy.split(',').forEach(bypass => {
+                this._httpProxyBypassHosts.push(new RegExp(bypass, 'i'));
```

Comment:
Don't the bypass needs to be processed a bit before turned into a regex?

Can think of a few scenarios

1. If there is a space after the comma it will now be part of the pattern
2. Escape . so that github.com does not match githubFcom
3. Normally adding a domain includes all subdomains e.g. having github.com should include raw.github.com

With this I'm thinking that maybe a version with safe checking of something like
`new RegExp('(^|.+\\.)' + bypass.trim().replace('.', '\\.'))`
Could work?

Reply:
I was also thinking about this, but i decided to align it with Line 128 
``` javascript
            if (requestOptions.proxy && requestOptions.proxy.proxyBypassHosts) {
                this._httpProxyBypassHosts = [];
                requestOptions.proxy.proxyBypassHosts.forEach(bypass => {
                    this._httpProxyBypassHosts.push(new RegExp(bypass, 'i'));
                });
            }
```

normally in linux environments the no_proxy is not a regex:

> This variable should contain a comma-separated list of domain extensions proxy should not be used for. For instance, if the value of no_proxy is ‘.mit.edu’, proxy will not be used to retrieve documents from MIT.

see http://www.gnu.org/software/wget/manual/html_node/Proxies.html